packages/components/pie-modal/test/component/pie-modal.spec.ts

```diff
@@ -26,779 +31,763 @@ test.describe('modal', () => {
+
+                    const focusedElement = await page.locator(':focus');
+                    const focusedElementId = await focusedElement.getAttribute('id');
+
+                    // Assert
+                    expect(focusedElementId).toBe('actual-focus');
+                });
+            });
+
+            test.describe('when not given', () => {
+                [{
+                    mechanism: 'close button',
+                    modalCloseFunction: async (page : Page) => {
+                        const modalPage = new PieModalPage(page);
+                        await modalPage.clickCloseModal();
+                    },
+                }, {
+                    mechanism: 'Esc key',
+                    modalCloseFunction: async (page : Page) => {
+                        await page.keyboard.press('Escape');
+                    },
+                }].forEach(({ mechanism, modalCloseFunction }) => {
+                    test.describe(`and closed by the ${mechanism}`, () => {
+                        test('should return focus to the element that opens the modal', async ({ page, mount }) => {
+                            // Arrange
+                            const component = renderTestPieModal({ isOpen: false });
+
+                            await mount(WebComponentTestWrapper, {
+                                props: {
+                                    pageMode: true,
+                                },
+                                slots: {
+                                    component,
+                                    pageMarkup: `<div>
+                                        <button id="not-me"></button>
+                                        <button data-test-id="open-modal" id="default"></button>
+                                    </div>`,
+                                },
                             });
-                        });
 
-                        // Act
-                        await page.click('[data-test-id="open-modal"]');
-                        await modalCloseFunction(page);
+                            await page.evaluate(() => {
+                                // Set up a button which opens the modal when clicked
+                                document.querySelector('[data-test-id="open-modal"]')?.addEventListener('click', () => {
+                                    document.querySelector('pie-modal')?.setAttribute('isOpen', 'true');
+                                });
+                            });
+
+                            // Act
+                            await page.click('[data-test-id="open-modal"]');
+                            await modalCloseFunction(page);
 
-                        const focusedElement = await page.locator(':focus');
-                        const focusedElementId = await focusedElement.getAttribute('id');
+                            const focusedElement = await page.locator(':focus');
```

Comment:
white space

Reply:
Some of these white spaces are from when wrapped all the tests in that modal.describe block.